pyro/contrib/autoguide/__init__.py

```diff
@@ -296,6 +304,22 @@ def median(self, *args, **kwargs):
         """
         return self(*args, **kwargs)
 
+    def covariance(self, *args, **kwargs):
+        """
+        Returns covariance of latent variables under Laplace (quadratic) approximation.
+        """
+        guide_trace = poutine.trace(self).get_trace(*args, **kwargs)
+        model_trace = poutine.trace(
+            poutine.replay(self.model, trace=guide_trace)).get_trace(*args, **kwargs)
+        loss = -model_trace.log_prob_sum()
+
+        latents = []
+        for _, site in guide_trace.iter_stochastic_nodes():
+            latents.append(site["value"])
+
+        H = _hessian(loss, latents)
+        return torch.inverse(H)
```

Comment:
@fritzo I keep this functionality in case some user (like me) wants to do Laplace approximation in constrained space. I hope that it is fine for you.

Reply:
Why don't you go further and make this into a method that returns a constrained `AutoMultivariateNormal` guide?

Reply:
I am afraid that it would be complicated. The `AutoMultivariateNormal` works on unconstrained variables. Things will be wrong if we use this `covariance` as the covariance of AutoMultivariateNormal class. In addition, as you have pointed out that this Laplace approximation can give wrong samples (not belong to the support), would it be better to avoid it? By this method, I just keep the most important statistics from Laplace approximation: the covariance. Users can use it to see the covariance, correlation, or get samples at their own risk.